### ISSUE TITLE
S4261: false diagnostic when using generic constraints

### ISSUE THREAD

**Issue by anreton** [SonarSource/sonar-dotnet#4799]
### Description
[S4261](https://rules.sonarsource.com/csharp/RSPEC-4261)

### Repro steps

```csharp
private TResult ExecuteAsync<TResult>(Func<Client, TResult> selector) where TResult : Task
{
	using (new OperationContextScope(this.client.InnerChannel))
	{
		return selector(this.client) ?? throw new InvalidOperationException();
	}
}
```

### Expected behavior

No false diagnostic for S4261.

### Actual behavior

Get false diagnostic for S4261.

### Related information

- Microsoft Visual Studio Community 2019 Version 16.11.0
- SonarAnalyzer.CSharp 8.27.0.35380 via Nuget

**Comment by dmitry-golovinov-sonarsource** [comment]
Hello @anreton! Thanks for leaving your feedback.  As it is stated in the RSPEC you've mentioned, this rule is deprecated, and will eventually be removed. We are not going to provide any support regarding deprecated rules. If you have activated it manually using ruleset or .editorconfig file, I recommend you to disable it.

**Comment by dmitry-golovinov-sonarsource** [comment]
Reopened the issue due to this thread https://github.com/SonarSource/sonar-dotnet/issues/4059
